ospfd/ospf_vty.c

```diff
@@ -5939,37 +5895,131 @@ static int show_ip_ospf_neighbor_int_detail_common(struct vty *vty,
 		}
 	}
 
-	if (use_json)
-		vty_json(vty, json);
-	else
+	if (use_json) {
+		if (!json_vrf)
+			vty_json(vty, json);
+	} else {
 		vty_out(vty, "\n");
+	}
 
 	return CMD_SUCCESS;
 }
 
-DEFUN (show_ip_ospf_neighbor_int_detail,
-       show_ip_ospf_neighbor_int_detail_cmd,
-       "show ip ospf neighbor IFNAME detail [json]",
-       SHOW_STR
-       IP_STR
-       "OSPF information\n"
-       "Neighbor list\n"
-       "Interface name\n"
-       "detail of all neighbors\n"
-       JSON_STR)
+DEFUN(show_ip_ospf_neighbor_int,
+      show_ip_ospf_neighbor_int_cmd,
+      "show ip ospf [vrf <NAME>] neighbor IFNAME [json]",
+      SHOW_STR IP_STR
+      "OSPF information\n"
+      VRF_CMD_HELP_STR
+      "Neighbor list\n"
+      "Interface name\n"
+      JSON_STR)
+{
```

Comment:
Please update the documentation as well with this change.

Reply:
Updated